Criterion/Report.hs

```diff
@@ -131,18 +150,11 @@ formatReport reports templateName = do
     -- If there were any issues during mustache template rendering, make sure
     -- to inform the user. See #127.
     forM_ warnings $ \warning -> do
-      -- The one thing we choose not to warn about is substituting in the `json`
-      -- key. The reason is that `json` is used in:
-      --
-      --   var reports = {{{json}}};
-      --
-      -- So `json` represents a raw JavaScript array. This is a bit skeevy by
-      -- mustache conventions, but redesigning the template to avoid this
-      -- warning would be more work than just substituting the array directly.
-      unless (warning == MustacheDirectlyRenderedValue (Key ["json"])) $
         criterionWarning $ displayMustacheWarning warning
     return formatted
   where
+    reportsJSON :: [Report] -> T.Text
+    reportsJSON = TL.toStrict . TL.concatMap escapeJSON . encodeToLazyText
```

Comment:
Nitpick: put a newline after `reportJSON` for consistency with other functions defined in this `where` clause.